src/vsr/replica.zig

```diff
@@ -7887,6 +7887,7 @@ pub fn ReplicaType(
                 .updating_checkpoint => |checkpoint| {
                     if (checkpoint.header.op == self.op_checkpoint()) {
                         self.sync_superblock_update_finish();
+                        assert(self.syncing == .idle);
```

Comment:
"Surely this is a waste of time and can't fire" thought I when adding this assert during debugging!